src/graphlib/graph.ts

```diff
@@ -24,23 +24,30 @@ var EDGE_KEY_DELIM = '\x01';
 //    edges up and, object properties, which have string keys, are the closest
 //    we're going to get to a performant hashtable in JavaScript.
 export class Graph {
-  _isDirected: boolean;
-  _isMultigraph: boolean;
-  _isCompound: boolean;
-  _label: any;
-  _defaultNodeLabelFn: () => any;
-  _defaultEdgeLabelFn: () => any;
-  _nodes: {};
-  _parent: {};
-  _children: {};
-  _in: {};
-  _preds: {};
-  _out: {};
-  _sucs: {};
-  _edgeObjs: {};
-  _edgeLabels: {};
-  _nodeCount = 0;
-  _edgeCount = 0;
+  protected _isDirected: boolean;
+  protected _isMultigraph: boolean;
+  protected _isCompound: boolean;
+  protected _label: any;
+  protected _defaultNodeLabelFn: () => any;
+  protected _defaultEdgeLabelFn: () => any;
+  protected _nodes: {};
+  protected _parent: {};
+  protected _children: {};
+  protected _in: {};
+  protected _preds: {};
+  protected _out: {};
+  protected _sucs: {};
+  protected _edgeObjs: {};
+  protected _edgeLabels: {};
+  /**
+   * Number of nodes in the graph. Should only be changed by the implementation. 
+   */
+  protected _nodeCount = 0;
+  /**
+   * Number of edges in the graph. Should only be changed by the implementation. 
+   */
+  protected _edgeCount = 0;
```

Comment:
- [private properties](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_properties) are properties that are private at the JavaScript language level. They can’t be accessed from outside.
- [TypeScript `private`](https://www.typescriptlang.org/docs/handbook/2/classes.html#private) fields are public fields, but TypeScript blocks access from outside the class. They are part of the public interface.
- [TypeScript `protected`](https://www.typescriptlang.org/docs/handbook/2/classes.html#protected) fields are fields that are like TypeScript `private` fields, except they may be used from subclasses.

I’m not dictating which one you should use, but you should understand the differences. I would personally pick real private properties.

Reply:
> I would personally pick real private properties.

I would normally pick that too for new code, but considering that this is code that we don't understand, and making variables like this properly `#private` in JavaScript would be a breaking change to third-party users, I think making it [TypeScript `private`](https://www.typescriptlang.org/docs/handbook/2/classes.html#private) (or [TypeScript `protected`](https://www.typescriptlang.org/docs/handbook/2/classes.html#protected) if we do extend `Graph`) is probably better.

If we ever make a v8 of this package, we could move these fields to proper `#private` variables.

In case you're curious as to what `@types/graphlib` does, apparently they're just ignored, so I guess this means that probably nobody is using these fields externally? https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e5048277e6e1d57cd5758ab630a04ef4810f3c98/types/graphlib/index.d.ts#L15

Reply:
Thanks for both of you for your feedback. I agree that using the javascript private field could break backward compatibility. Out of laziness I left `protected` but `private` would work.